Added evaluation script for qualcomm LlamaModel

rohansjoshi opened the pull request:
Summary:
Script for evaluating models which follow qualcomm's LlamaModel definition, on lm eval harness tasks such as WikiText

Results for WikiText evaluation task:

| Model Name | max_seq_len | word_perplexity
|----------|----------|----------|
| Llama 1B Instruct  | 128   | 34.82890030691187 |
| Llama 1B Instruct  | 512   | 22.919538703371582 |

Differential Revision: D76634688

cccclai reviewed (approved):
Thank you for adding the eval scripts. @shewu-quic  @haowhsu-quic fyi we're adding eval scripts here and trying to improve the accuracy for ptq

haowhsu-quic commented:
> Thank you for adding the eval scripts. @shewu-quic @haowhsu-quic fyi we're adding eval scripts here and trying to improve the accuracy for ptq

Thank you, this is very helpful!

rohansjoshi committed:
Added evaluation script for qualcomm LlamaModel (#11663)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/11663

Script for evaluating models which follow qualcomm's LlamaModel definition, on lm eval harness tasks such as WikiText

Results for WikiText evaluation task:

| Model Name | max_seq_len | word_perplexity
|----------|----------|----------|
| Llama 1B Instruct  | 128   | 34.82890030691187 |
| Llama 1B Instruct  | 512   | 22.919538703371582 |

Reviewed By: cccclai

Differential Revision: D76634688
